client/lib/happychat/connection.js

```diff
@@ -2,51 +2,62 @@
  * External dependencies
  */
 import IO from 'socket.io-client';
-import { EventEmitter } from 'events';
-import config from 'config';
 import { v4 as uuid } from 'uuid';
 
 /**
  * Internal dependencies
  */
-import { HAPPYCHAT_MESSAGE_TYPES } from 'state/happychat/constants';
+import {
+	HAPPYCHAT_MESSAGE_TYPES
+} from 'state/happychat/constants';
+import {
+	receiveChatEvent,
+	requestChatTranscript,
+	setConnected,
+	setConnecting,
+	setDisconnected,
+	setHappychatAvailable,
+	setHappychatChatStatus,
+	setReconnecting,
+} from 'state/happychat/actions';
 
-/*
- * Happychat client connection for Socket.IO
- */
 const debug = require( 'debug' )( 'calypso:happychat:connection' );
 
-class Connection extends EventEmitter {
+const isString = ( socket ) => ( typeof socket === 'string' || socket instanceof String );
+const buildConnection = ( socket ) => isString( socket )
+	? new IO( socket ) // If socket is an URL, connect to server.
+	: socket; // If socket is not an url, use it directly. Useful for testing.
+
+class Connection {
```

Comment:
I'm curios if there was a good reason for using `EventEmitter` and re-emitting the events. However this definitely looks cleaner. Thanks for doing it.